rosbag2_transport/test/rosbag2_transport/test_qos.cpp

```diff
@@ -95,3 +95,148 @@ TEST(TestQoS, detect_new_qos_fields)
   };
   EXPECT_EQ(profile.history, RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT);  // fix "unused variable"
 }
+
+class AdaptiveQoSTest : public ::testing::Test
+{
+public:
+  AdaptiveQoSTest() = default;
+
+  rclcpp::TopicEndpointInfo make_endpoint(const rclcpp::QoS & qos)
+  {
+    rcl_topic_endpoint_info_t endpoint_info {
+      "some_node_name",
+      "some_node_namespace",
+      "some_topic_type",
+      RMW_ENDPOINT_PUBLISHER,
+      {0},
+      qos.get_rmw_qos_profile(),
+    };
+
+    return rclcpp::TopicEndpointInfo(endpoint_info);
+  }
+
+  void add_endpoint(const rclcpp::QoS & qos)
+  {
+    endpoints_.push_back(make_endpoint(qos));
+  }
+
+  const std::string topic_name_{"/topic"};
+  std::vector<rclcpp::TopicEndpointInfo> endpoints_{};
+  const rosbag2_transport::Rosbag2QoS default_offer_{};
+  const rosbag2_transport::Rosbag2QoS default_request_{};
+};
+
+TEST_F(AdaptiveQoSTest, adapt_request_empty_returns_default)
+{
+  using rosbag2_transport::Rosbag2QoS;
```

Comment:
Why not move this out of the individual test scope and put it just before the test fixture?